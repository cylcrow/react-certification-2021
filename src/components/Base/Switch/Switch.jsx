import React from 'react';
import styled from 'styled-components';

const StyledInput = styled.input``;

const Switch = () => {
  return <StyledInput type="checkbox" />;
};

export default Switch;
