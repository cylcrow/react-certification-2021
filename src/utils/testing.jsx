import React from "react";
import { render } from '@testing-library/react';
import { ThemeProvider } from "styled-components";
import { lightTheme } from '../providers/theme/themes';

export const renderWithTheme = (Component, theme = lightTheme) => {
    return render(
        <ThemeProvider theme={{ theme, switchTheme: jest.fn() }}>
        { Component }
        </ThemeProvider>
    );
}