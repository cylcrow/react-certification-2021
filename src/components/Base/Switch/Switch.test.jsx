import React from 'react';
import '../../../utils/testing';
import 'jest-styled-components';
import { render, unmountComponentAtNode } from 'react-dom';
import toJson from 'enzyme-to-json';
// import { mount, shallow } from 'enzyme';
import Switch from './Switch';

let container;
const build = () => {
  render(<Switch />, container);
};

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
  return container;
});

describe('Switch', () => {
  it('renders', () => {
    const wrapper = build(<Switch />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

// describe('Switch styles and props', () => {
//   it('applies default styling', () => {
//     const tree = mount(<Switch />);
//   });
// });
