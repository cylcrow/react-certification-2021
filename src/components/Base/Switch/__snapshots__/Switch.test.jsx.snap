// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch renders 1`] = `undefined`;
